starlite/middleware/session/server_side.py

```diff
@@ -0,0 +1,147 @@
+import secrets
+from typing import Any, Dict, Type
+
+from starlite import ASGIConnection, Cookie
+from starlite.datastructures import MutableScopeHeaders
+from starlite.middleware.session.base import BaseBackendConfig, BaseSessionBackend
+from starlite.storage.base import StorageBackend
+from starlite.types import Empty, Message, ScopeSession
+
+
+class ServerSideBackend(BaseSessionBackend["ServerSideSessionConfig"]):
+    """Base class for server-side backends.
+
+    Implements :class:`BaseSessionBackend` and defines and interface which subclasses can
+    implement to facilitate the storage of session data.
+    """
+
+    __slots__ = ("storage",)
+
+    def __init__(self, config: "ServerSideSessionConfig") -> None:
+        """Initialize ``ServerSideBackend``
+
+        Args:
+            config: A subclass of ``ServerSideSessionConfig``
+        """
+        super().__init__(config=config)
+        self.storage = config.storage
+
+    async def get(self, session_id: str) -> bytes | None:
+        """Retrieve data associated with ``session_id``.
+
+        Args:
+            session_id: The session-ID
+
+        Returns:
+            The session data, if existing, otherwise ``None``.
+        """
+        return await self.storage.get(session_id)
+
+    async def set(self, session_id: str, data: bytes) -> None:
+        """Store ``data`` under the ``session_id`` for later retrieval.
+
+        If there is already data associated with ``session_id``, replace
+        it with ``data`` and reset its expiry time
+
+        Args:
+            session_id: The session-ID
+            data: Serialized session data
+
+        Returns:
+            None
+        """
+        await self.storage.set(session_id, data)
+
+    async def delete(self, session_id: str) -> None:
+        """Delete the data associated with ``session_id``. Fails silently if no such session-ID exists.
+
+        Args:
+            session_id: The session-ID
+
+        Returns:
+            None
+        """
+        await self.storage.delete(session_id)
+
+    def generate_session_id(self) -> str:
+        """Generate a new session-ID, with
+        n=:attr:`session_id_bytes <ServerSideSessionConfig.session_id_bytes>` random bytes.
+
+        Returns:
+            A session-ID
+        """
+        return secrets.token_hex(self.config.session_id_bytes)
+
+    async def store_in_message(
+        self, scope_session: "ScopeSession", message: "Message", connection: ASGIConnection
```

Comment:
> or if we no longer need to quote - remove the quotes from other types

I used the `__future__.annotations` for new stuff but I didn't want to introduce additional changes in other parts, so it might be done in a single PR. But we can also start doing it progressively.

Reply:
i would rather we do a PR that does this for the entire code base rahter than go for progressive adoption - but you can of course do this in this scope.

Reply:
I've been doing same with the new modules I've added in the DTO refactor as well.

How about we make the annotations clean up first thing after 2.0 is done, and progressively add them in the meantime?

Reply:
Could do that as well. It’s not a breaking change anyway, so no need to do it in 2.0.